raffle.js

```diff
@@ -1,41 +1,158 @@
 #!/usr/bin/env node
 
-var fs = require('fs');
+'use strict'
 
-var name = 'raffle.json';
-var noop = function() {};
+const fs = require('fs')
+const path = require('path')
+const blessed = require('blessed')
+const contrib = require('blessed-contrib')
+const screen = blessed.screen()
+const grid = new contrib.grid({rows: 2, cols: 5, screen: screen})
 
-function random(count){
+const argv = process.argv
 
-  read(function(err, data) {
-    var i = 0;
-    var arr = [];
-    while(i < count){
-      var a = (Math.random()*600).toFixed(0);
-      if(!~data.indexOf(a) && !~arr.indexOf(a)){
-        arr.push(a);
-        i++;
-      }
-    }
-    console.log('%d total \n%s', arr.length, arr.join(','));
-    write(data.concat(arr), noop);
-  })
+const name = 'raffle.json'
+const file = path.resolve(__dirname, name)
+const total = parseInt(argv[2]) || 800
+
+const store = {
+  get luckies() {
+    return read(file)
+  }
 }
 
-function write(data, cb) {
-  fs.writeFile(name, JSON.stringify(data), cb);
+const probability = (C) => {
+  return (x) => {
+    return parseFloat((1 / (1 + C * Math.tan(x / 5))).toFixed(4))
+  }
 }
 
-function read(cb) {
-  fs.readFile(name, function(err, data) {
-    if (err) {
-      if (err.code == 'ENOENT') return cb(null, ["0"]);
-      return cb(err);
+const weight = probability(total);
+
+const write = (luckies, lucky) => {
+  luckies = luckies.concat(lucky).reduce((p, n) => {
+    const lucky = p.filter(l => l.number === n.number)[0]
+    if (lucky) {
+      p = p.filter(l => l.number !== n.number)
     }
-    cb(null, JSON.parse(data.toString()));
-  })
+    n.times = lucky ? lucky.times ++ : (n.times || 1)
+    delete n.weight
+    p.push(n)
+    return p
+  }, [])
+  fs.writeFileSync(file, JSON.stringify(luckies))
 }
 
-var argv = process.argv;
-var count = argv[2];
-random(count || 0);
+const read = (f) => {
+  if (!fs.statSync(f).isFile()) {
+    return []
+  }
+  return JSON.parse(fs.readFileSync(f).toString())
+    .map(l => {
+      const wei = weight(l.times)
+      l.weight = wei > 0 ? wei : 0
+      return l
+    })
+}
+
+const random = (luckies, length) => {
+  let sum = 0,
+    res = [],
+    wei = weight(1);
+
+  for (var l in luckies) {
+    sum += luckies[l].weight
+  }
+  sum += (total - luckies.length) * wei
+  for (var i = 1; i <= total; i ++) {
+    const lucky = luckies.filter(l => l.number === i)[0] || {}
+    const w = lucky.weight === 0 ? 0 : ((lucky.weight || wei) + Math.random() * sum)
+    res.push(w)
+  }
+  return res.map((w, i) => {
+      return {
+        number: i,
+        weight: w
+      }
+    })
+    .sort((a, b) => b.weight - a.weight)
+    .map(l => {
+      delete l.weight
+      return l
+    })
+    .slice(0, length || 1)
+}
+
+// Probability Graph
+const line = grid.set(0, 2, 1, 2, contrib.line, {
```

Comment:
Maybe you can separate the function to generate number and draw graph into different files so it's easy to read the code?